benches/main.rs

```diff
@@ -0,0 +1,106 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use event_manager::utilities::subscribers::{
+    CounterInnerMutSubscriber, CounterSubscriber, CounterSubscriberWithData,
+};
+use event_manager::{EventManager, EventSubscriber, MutEventSubscriber, SubscriberOps};
+use std::sync::{Arc, Mutex};
+
+pub fn run_basic_subscriber(c: &mut Criterion) {
```

Comment:
Looks like the only difference between `run_basic_subscriber`, `run_arc_mutex_subscriber`, `run_subscriber_with_inner_mut`, and `run_multiple_subscriber_types` is the kind of dispatch/synchronization used by the subscriber, which is ultimately a subscriber implementation detail. I think that, for the purpose of benchmarking the event manager, we should only use the simplest subscriber type, so we add the least amount of noise to the operation of the event manager itself. Moreover, all 200 subscribers being active at the same time is a bit extreme; I would also add a test where many subscribers are registered, but only one is triggered.

Reply:
The reason for having multiple subscribers types here is that it also helps the user of this crate to see what is the performance impact of things like dynamic dispatch and Mutex. This way people can take a more informed decision when developing a component around the event-manager. I would also like us to improve the current benchmark infrastructure so that we can have visualizations. Visualization the differences between subscriber types is not only super cool, but again should help people take informed decisions. With that in mind, I would like to disagree with the proposal of only testing the basic subscriber type.

Getting to the 200 subscribers being active at the same time, the goal is also to see how the event manager behaves under stress conditions. I would say it is safe to say that if it handles well 200 events at a time, that it would do just fine with 1 event. Nevertheless, I will add a test where only one subscriber has active events.